client/src/main/java/io/pravega/client/connection/impl/TcpClientConnection.java

```diff
@@ -115,7 +113,7 @@ public void run() {
                     WireCommand command = readCommand(in, buffer);
                     if (command instanceof WireCommands.DataAppended) {
                         WireCommands.DataAppended dataAppended = (WireCommands.DataAppended) command;
-                        batchSizeTracker.recordAck(dataAppended.getEventNumber());
+                        flowToBatchSizeTracker.getAppendBatchSizeTrackerByFlowId(dataAppended.getRequestId()).recordAck(dataAppended.getEventNumber());
```

Comment:
Using `io.pravega.client.connection.impl.Flow#toFlowID` api to get the requestId will ensure  only the flow id is used. This will ensure requests via the RawClient do not create multiple `AppendBatchSizeTrackers`.

Reply:
Done.